lib/src/server/service.dart

```diff
@@ -56,36 +56,44 @@ class ServiceMethod<Q, R> {
         return handler(call, _toSingleFuture(requests));
       }
     } else {
-      Future<R> response;
+      Future<R>? response;
       if (streamingRequest) {
         response = handler(call, requests);
       } else {
         response = handler(call, _toSingleFuture(requests));
       }
-      return response.asStream();
+      return response!.asStream();
     }
   }
 
   Future<Q> _toSingleFuture(Stream<Q> stream) {
-    Q _ensureOnlyOneRequest(Q previous, Q element) {
+    Q? _ensureOnlyOneRequest(Q? previous, Q? element) {
```

Comment:
`Q element` and then below remove `cast` and instead `fold<Q?>`

Reply:
Done